fe/fe-core/src/main/java/org/apache/doris/catalog/PartitionInfo.java

```diff
@@ -465,13 +465,15 @@ public void readFields(DataInput in) throws IOException {
             }
 
             idToInMemory.put(partitionId, in.readBoolean());
-            if (Config.isCloudMode()) {
-                // HACK: the origin implementation of the cloud mode has code likes:
-                //
-                //     idToPersistent.put(partitionId, in.readBoolean());
-                //
-                // keep the compatibility here.
-                in.readBoolean();
+            if (Env.getCurrentEnvJournalVersion() > FeMetaVersion.VERSION_129) {
+                if (Config.isCloudMode()) {
+                    // HACK: the origin implementation of the cloud mode has code likes:
+                    //
+                    //     idToPersistent.put(partitionId, in.readBoolean());
+                    //
+                    // keep the compatibility here.
+                    in.readBoolean();
+                }
```

Comment:
I am confused by code above.

Reply:
Same as above, If we backup data from a 2.1 doris and restore data to a 3.0 cloud doris, it will fail at read BackupMeta.

The version 129 is the last meta version of 2.1.

Reply:
add the above comment to this code, confused too...